### PR TITLE
Implement GCJ style

### PR DESCRIPTION
**Pull request by aguss787**

### CODE REVIEW

**Review by fushar — CHANGES_REQUESTED** [review]

File: judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/updater/ContestScoreboardUpdaterDispatcher.java [hunk 1]
```diff
@@ -36,6 +36,8 @@ public void run() {
     }
 
     public synchronized void updateAsync(Contest contest) {
+        LOGGER.info(contest.getName());
```
[the remark this answers is not in the document]
Comment (aguss787) [reply judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/updater/ContestScoreboardUpdaterDispatcher.java]: Whoops, it is not supposed to be there

File: judgels-backends/uriel/uriel-app/src/integTest/java/judgels/uriel/api/contest/problem/ContestProblemServiceIntegrationTests.java [hunk 2]
```diff
@@ -79,12 +79,14 @@ void end_to_end_flow() {
                         .problemJid(PROBLEM_1_JID)
                         .status(OPEN)
                         .submissionsLimit(10)
+                        .points(0)
```
[the remark this answers is not in the document]
Comment (aguss787) [reply judgels-backends/uriel/uriel-app/src/integTest/java/judgels/uriel/api/contest/problem/ContestProblemServiceIntegrationTests.java:82]: No, even though it is optional, the return value will have points equals to zero. Hence, we will still need to set points to zero

File: judgels-frontends/raphael/src/routes/uriel/contests/single/problems/modules/editor/defaultContestProblemEditor.test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import defaultContestProblemEditor from './defaultContestProblemEditor';
+
+describe('contestProblemValidations', () => {
```
Comment (fushar) [inline judgels-frontends/raphael/src/routes/uriel/contests/single/problems/modules/editor/defaultContestProblemEditor.test.ts]:
```suggestion
describe('defaultContestProblemEditor', () => {
```

File: judgels-frontends/raphael/src/routes/uriel/contests/single/problems/modules/editor/defaultContestProblemEditor.test.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  test('ValidProblemsSetData', () => {
```
Comment (fushar) [inline judgels-frontends/raphael/src/routes/uriel/contests/single/problems/modules/editor/defaultContestProblemEditor.test.ts]:
```suggestion
  test('validator', () => {
```

File: judgels-frontends/raphael/src/routes/uriel/contests/single/problems/ContestProblemEditDialog/ContestProblemEditDialog.tsx [hunk 5]
```diff
@@ -34,6 +44,18 @@ export class ContestProblemEditDialog extends React.Component<
     );
   }
 
+  componentDidUpdate(prevProps: ContestProblemEditDialogProps) {
+    if (prevProps.contest.style !== this.props.contest.style) {
+      this.updateComponents();
+    }
+  }
+
+  private updateComponents() {
```
Comment (fushar) [inline judgels-frontends/raphael/src/routes/uriel/contests/single/problems/ContestProblemEditDialog/ContestProblemEditDialog.tsx]:
```suggestion
  private updateEditor() {
```

File: judgels-frontends/raphael/src/routes/uriel/contests/single/problems/ContestProblemEditDialog/ContestProblemEditDialog.tsx [hunk 6]
```diff
@@ -34,6 +44,18 @@ export class ContestProblemEditDialog extends React.Component<
     );
   }
 
+  componentDidUpdate(prevProps: ContestProblemEditDialogProps) {
+    if (prevProps.contest.style !== this.props.contest.style) {
+      this.updateComponents();
```
Comment (fushar) [inline judgels-frontends/raphael/src/routes/uriel/contests/single/problems/ContestProblemEditDialog/ContestProblemEditDialog.tsx]:
```suggestion
      this.updateEditor();
```

**Review by fushar — CHANGES_REQUESTED** [review]

File: judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/updater/ContestScoreboardUpdater.java [hunk 7]
```diff
@@ -65,21 +67,28 @@ public void update(Contest contest) {
...
 
-        List<Submission> submissions = submissionStore.getSubmissions(
+        List<Submission> submissions = new ArrayList<>(submissionStore.getSubmissions(
                 contest.getJid(),
                 Optional.empty(),
                 Optional.empty(),
                 Optional.empty())
-                .getPage();
+                .getPage());
+        submissions.sort(Comparator.comparingLong(Submission::getId));
```
[the remark this answers is not in the document]
Comment (aguss787) [reply judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/updater/ContestScoreboardUpdater.java:91]: Apparently it is not :/.
the list is reversed when I try it

**Review by fushar — APPROVED** [review]

**Pull request merged**